File: Codes/2302-count-subarrays-with-score-less-than-k.java
/*
 * @Author: LetMeFly
 * @Date: 2025-04-29 23:46:43
 * @LastEditors: LetMeFly.xyz
 * @LastEditTime: 2025-04-30 10:41:19
 */
class Solution {
    public long countSubarrays(int[] nums, long k) {
        long ans = 0, cnt = 0;
        for (int l = 0, r = 0; r < nums.length; r++) {
            cnt += nums[r];
            while (cnt * (r - l + 1) >= k) {
                cnt -= nums[l++];
            }
            ans += (r - l + 1);
        }
        return ans;
    }
}

File: Codes/2302-count-subarrays-with-score-less-than-k.py
'''
Author: LetMeFly
Date: 2025-04-29 23:46:38
LastEditors: LetMeFly.xyz
LastEditTime: 2025-04-30 10:39:13
'''
from typing import List

class Solution:
    def countSubarrays(self, nums: List[int], k: int) -> int:
        ans = cnt = l = 0
        for r in range(len(nums)):
            cnt += nums[r]
            while cnt * (r - l + 1) >= k:
                cnt -= nums[l]
                l += 1
            ans += (r - l + 1)
        return ans